### PR TITLE
added service principal, pin AML version

### PR DESCRIPTION
**Pull request by fboylu**

### CODE REVIEW

File: utilities.py [hunk 1]
```diff
@@ -82,3 +87,26 @@ def read_questions(path, id, answerid):
     questions = questions.set_index(id, drop=False)
     questions.sort_index(inplace=True)
     return questions
+
+def _get_auth(env_path):
```
Comment (msalvaris) [inline utilities.py]: @fboylu Can you change the signature from private (_get_auth) to public (get_auth)
In Python functions that start with underscore are considered private and should only be used inside the Python module.
Sorry I should have mentioned that I was using it locally inside my Python module

**Comment by msalvaris** [comment]
@grecoe has code for the az login so that should be fine

**Comment by msalvaris** [comment]
@fboylu said az cli is isntalled as a pip dependency in env

File: utilities.py (hunk 1 above)
Comment (fboylu) [reply utilities.py]: @msalvaris this is fixed now. Thanks.

**Pull request merged**